src/homekit.cs

```diff
@@ -1416,4 +1528,211 @@ interface HMCameraAudioControl
 		[NullAllowed, Export ("volume", ArgumentSemantic.Strong)]
 		HMCharacteristic Volume { get; }
 	}
+
+	[Watch (4,0), TV (11,0), iOS (11,0)]
+	[BaseType (typeof(HMTimeEvent))]
+	[DisableDefaultCtor]
+	interface HMCalendarEvent : NSCopying, NSMutableCopying {
+
+		[Export ("initWithFireDateComponents:")]
+		IntPtr Constructor (NSDateComponents fireDateComponents);
+
+		[Export ("fireDateComponents", ArgumentSemantic.Strong)]
+		NSDateComponents FireDateComponents { get; }
+	}
+
+	[Watch (4,0), TV (11,0), iOS (11,0)]
+	[BaseType (typeof(HMCalendarEvent))]
+	[DisableDefaultCtor]
+	interface HMMutableCalendarEvent {
+
+		[Export ("initWithFireDateComponents:")]
+		IntPtr Constructor (NSDateComponents fireDateComponents);
+
+		[Export ("fireDateComponents", ArgumentSemantic.Strong)]
+		NSDateComponents FireDateComponents { get; set; }
```

Comment:
don't you get warnings from this ? c# does not allow overriding properties that don't match (no setter vs setter)

the non-mutable version should have both a getter and a setter, the later should be decorated with `[NotImplemented]`. That solves the issue and makes the code work like expected.